apps/website/docs/azure/architecting/eventgrid-storage-functions.md

```diff
@@ -0,0 +1,434 @@
+  parseBlob,
+  toAzureFunctionHandler,
+);
+
+const parseBlob = ({ inputs }) =>
+  pipe(
+    inputs[0] as Buffer, // Blob content received directly
+    (blob) => blob.toString("utf-8"),
+    parseJson,
+    TE.fromEither,
+    TE.chainW((decodedItem) => processItem(decodedItem)),
+  );
+```
+
+</TabItem>
+<TabItem value="after" label="After (EventGrid)">
+
+```typescript
+import { BlobServiceClient } from "@azure/storage-blob";
+import { DefaultAzureCredential } from "@azure/identity";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+
+export const onBlobEventEntryPoint = pipe(
+  onBlobEventHandler(),
+  parseEventGridEvent,
+  toAzureFunctionHandler,
+);
+
+const parseEventGridEvent = ({ inputs }) =>
+  pipe(
+    inputs[0] as EventGridEvent, // Event metadata only
+    extractBlobInfo,
+    TE.fromEither,
+    TE.chainW(fetchAndProcessBlob),
+  );
+
+const extractBlobInfo = (event: EventGridEvent) =>
+  E.tryCatch(
+    () => ({
+      containerName: event.subject.split("/")[6],
+      blobName: event.subject.split("/").slice(8).join("/"),
+    }),
+    (e) => new Error(`Failed to extract blob info: ${e}`),
+  );
+
+const fetchAndProcessBlob = (blobInfo: BlobInfo) =>
+  pipe(
+    TE.tryCatch(
+      async () => {
+        const credential = new DefaultAzureCredential();
+        const blobServiceClient = new BlobServiceClient(
+          `https://${process.env.STORAGE_ACCOUNT_NAME}.blob.core.windows.net`,
+          credential,
+        );
+        const blobClient = blobServiceClient
+          .getContainerClient(blobInfo.containerName)
+          .getBlobClient(blobInfo.blobName);
+
+        const downloadResponse = await blobClient.download();
+        const content = await streamToString(
+          downloadResponse.readableStreamBody,
+        );
+        return JSON.parse(content);
+      },
+      (e) => new Error(`Failed to fetch blob: ${e}`),
+    ),
+    TE.chainW(processItem),
+  );
+
+const streamToString = async (
+  stream: NodeJS.ReadableStream,
+): Promise<string> => {
+  const chunks: Buffer[] = [];
+  for await (const chunk of stream) chunks.push(Buffer.from(chunk));
+  return Buffer.concat(chunks).toString("utf-8");
+};
+```
+
+**Key changes:**
+
+- EventGrid provides event metadata, not blob content
+- Must fetch blob using Azure Storage SDK with managed identity
+- Use `DefaultAzureCredential` for RBAC authentication
+
+</TabItem>
+</Tabs>
+
+### Step 3: Update function.json
+
+<Tabs>
+<TabItem value="before" label="Before (Blob Trigger)">
+
+```json
+{
+  "bindings": [
+    {
+      "name": "blobContent",
+      "type": "blobTrigger",
+      "direction": "in",
+      "path": "%CONTAINER_NAME%/{name}",
+      "connection": "STORAGE_CONNECTION_STRING"
+    }
+  ],
+  "scriptFile": "../dist/main.js",
+  "entryPoint": "onBlobChangeEntryPoint"
+}
+```
+
+</TabItem>
+<TabItem value="after" label="After (EventGrid)">
+
+```json
```

Comment:
issue: this a deprecated pattern which is used in the former Azure Function runtime, we must use v4 model in our examples